pkg/api/install/install.go

```diff
@@ -42,20 +42,32 @@ var accessor = meta.NewAccessor()
 var availableVersions = []unversioned.GroupVersion{v1.SchemeGroupVersion}
 
 func init() {
+	registered.RegisterVersions(availableVersions...)
+
 	externalVersions := []unversioned.GroupVersion{}
-	for _, allowedVersion := range registered.GroupVersionsForGroup(api.GroupName) {
-		for _, externalVersion := range availableVersions {
-			if externalVersion == allowedVersion {
-				externalVersions = append(externalVersions, externalVersion)
-			}
+	for _, v := range availableVersions {
+		if registered.IsAllowedVersion(v) {
+			externalVersions = append(externalVersions, v)
 		}
 	}
-
 	if len(externalVersions) == 0 {
 		glog.V(4).Infof("No version is registered for group %v", api.GroupName)
 		return
 	}
+	if err := registered.EnableVersions(externalVersions...); err != nil {
```

Comment:
@nikhiljindal, per our offline discussion, the order is still marking those versions as enabled first, and then actually enable the versions, because we want to check we only add the enabled versions to scheme.